homeassistant/components/firmata/config_flow.py

```diff
@@ -0,0 +1,39 @@
+"""Config flow to configure firmata component."""
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+
+from .const import CONF_SERIAL_PORT, DOMAIN
+
+
+@callback
+def configured_boards(hass):
+    """Return a set of all configured boards."""
+    return {
+        entry.data[CONF_NAME]: entry
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class FirmataFlowHandler(config_entries.ConfigFlow):
+    """Handle a firmata config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_import(self, import_config):
+        """Import a firmata board as a config entry.
+
+        This flow is triggered by `async_setup` for configured boards.
+
+        This will execute for any board that does not have a
+        config entry yet (based on entry_id).
+        """
+        name = f"serial-{import_config[CONF_SERIAL_PORT]}"
+        import_config[CONF_NAME] = name
+
+        return self.async_create_entry(
```

Comment:
We normally check the connection to the device before creating the entry.

Reply:
Does this mean calling `get_board()` at this spot in the config flow? The only way check for a connection is to initialize the board. Is the YAML config validated at this point? How do I then pass the Pymata instance to `async_setup_entry()` from the config flow? Can I pass more arguments?

I suppose it doesn't ultimately make a difference on the user end where the board is connected/errors in setup since it is configured entirely in YAML, but I understand consistency is important.

Reply:
We normally connect twice, once in the config flow and once when setting up the config entry. We don't keep any state from the config flow.

The yaml config schema can only validate the structure and type, not if we can connect or not.

Reply:
OK, so I will initialize and then shutdown the board in the config flow and then do it again.